src/commands/price.ts

```diff
@@ -0,0 +1,186 @@
+		);
+		const nameSubcommand = buildLocalisedCommand(
+			new SlashCommandSubcommandBuilder(),
+			() => c("command-option").t`name`,
+			() => c("command-option-description").t`Display the price for the card with this name.`
+		);
+		const nameOption = buildLocalisedCommand(
+			new SlashCommandStringOption().setRequired(true),
+			() => c("command-option").t`input`,
+			() => c("command-option-description").t`Card name, fuzzy matching supported.`
+		);
+		const passwordSubcommand = buildLocalisedCommand(
+			new SlashCommandSubcommandBuilder(),
+			() => c("command-option").t`password`,
+			() => c("command-option-description").t`Display the price for the card with this password.`
+		);
+		const passwordOption = buildLocalisedCommand(
+			new SlashCommandStringOption().setRequired(true),
+			() => c("command-option").t`input`,
+			() =>
+				c("command-option-description")
+					.t`Card password, the eight-digit number printed on the bottom left corner.`
+		);
+		const konamiIdSubcommand = buildLocalisedCommand(
+			new SlashCommandSubcommandBuilder(),
+			() => c("command-option").t`konami-id`,
+			() => c("command-option-description").t`Display the price for the card with this official database ID.`
+		);
+		const konamiIdOption = buildLocalisedCommand(
+			new SlashCommandStringOption().setRequired(true),
+			() => c("command-option").t`input`,
+			() => c("command-option-description").t`Konami's official card database identifier.`
+		);
+		const vendorOption = buildLocalisedCommand(
+			new SlashCommandStringOption().setRequired(true),
+			() => c("command-option").t`vendor`,
+			() => c("command-option-description").t`The vendor to fetch the price data from.`
+		).addChoices(...Object.entries(CHOICES_GLOBAL).map(([value, name]) => buildLocalisedChoice(value, name)));
+		nameSubcommand
+			.addStringOption(nameOption)
+			.addStringOption(vendorOption)
+			.addStringOption(getInputLangStringOption());
+		passwordSubcommand.addStringOption(passwordOption).addStringOption(vendorOption);
+		konamiIdSubcommand.addStringOption(konamiIdOption).addStringOption(vendorOption);
+		builder.addSubcommand(nameSubcommand).addSubcommand(passwordSubcommand).addSubcommand(konamiIdSubcommand);
+		return builder.toJSON();
+	}
+
+	protected override get logger(): Logger {
+		return this.#logger;
+	}
+
+	// TODO: i18n
+	private vendorFormats = {
+		tcgplayer: (price: number): string => `$${price.toFixed(2)}`,
+		cardmarket: (price: number): string => `€${price.toFixed(2)}`,
+		coolstuffinc: (price: number): string => `$${price.toFixed(2)}`
+	};
+
+	async getPrice(card: Static<typeof CardSchema>, vendor: string): Promise<APIPrice | undefined> {
+		if (!card.name.en) {
+			// note: this is about our database, not the user input. Could clarify.
+			// unsure where, how or if to localise this error. Will it even come up?
+			throw new Error(t`Sorry, I can't find the price for a card with no English name!`);
+		}
```

Comment:
We find out in a new issue :) like the CardMarket 0 that this API provides and the currency format localisation. Out of scope